snc.sh

```diff
@@ -191,7 +191,8 @@ retry ${OC} scale --replicas=1 ingresscontroller/default -n openshift-ingress-op
 retry ${OC} patch config.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
 
 # Add a tip in the login page
-retry ${OC} get secrets -n openshift-authentication v4-0-config-system-ocp-branding-template -o json | ${JQ} -r '.data["login.html"]'  | base64 -d > login.html
+retry ${OC} get secrets -n openshift-authentication v4-0-config-system-ocp-branding-template -o json > tmp_secret_template.json
+${JQ} -r '.data["login.html"]' tmp_secret_template.json | base64 -d > login.html
```

Comment:
Wondering if
```diff
diff --git a/snc.sh b/snc.sh
index 4ffe9ad..d3dfaef 100755
--- a/snc.sh
+++ b/snc.sh
@@ -191,7 +191,7 @@ retry ${OC} scale --replicas=1 ingresscontroller/default -n openshift-ingress-op
 retry ${OC} patch config.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
 
 # Add a tip in the login page
-retry ${OC} get secrets -n openshift-authentication v4-0-config-system-ocp-branding-template -o json | ${JQ} -r '.data["login.html"]'  | base64 -d > login.html
+retry_xx /dev/null ${OC} get secrets -n openshift-authentication v4-0-config-system-ocp-branding-template -o json | ${JQ} -r '.data["login.html"]'  | base64 -d > login.html
 ${PATCH} login.html < login.html.patch
 retry ${OC} create secret generic login-template --from-file=login.html -n openshift-config
 
diff --git a/tools.sh b/tools.sh
index 288bd74..2cad5c2 100755
--- a/tools.sh
+++ b/tools.sh
@@ -88,7 +88,9 @@ if ! which ${PATCH}; then
     sudo yum -y install /usr/bin/patch
 fi
 
-function retry {
+function retry_xx {
+    local output=$1
+    shift
     local retries=10
     local count=0
     until "$@"; do
@@ -106,6 +108,10 @@ function retry {
     return 0
 }
 
+function retry {
+        retry /dev/stdout $@
+}
+
 function get_vm_prefix {
     local crc_vm_name=$1
     # This random_string is created by installer and added to each resource type,
```
could work.

Alternatively, have you tried:
```
branding_template=$(retry ${OC} get secrets -n openshift-authentication v4-0-config-system-ocp-branding-template -o json)
${JQ} ...
```

Actually I'm not sure I understand why the pipe won't work when the redirect does :-/
branding_template

Reply:
> Actually I'm not sure I understand why the pipe won't work when the redirect does :-/
```bash
#!/bin/bash

set -exuo pipefail

export LC_ALL=C.UTF-8
export LANG=C.UTF-8

function retry {
    local retries=4
    local count=0
    until "$@"; do
        exit=$?
        wait=$((2 ** $count))
        count=$(($count + 1))
        if [ $count -lt $retries ]; then
            echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
            sleep $wait
        else
            echo "Retry $count/$retries exited $exit, no more retries left."
            return $exit
        fi
    done
    return 0
}

function s_grep {
	echo "Hello there"
	false
}

retry false > tmp_file
```